components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/OAuth2ScopeService.java

```diff
@@ -306,6 +324,350 @@ public Scope updateScope(Scope updatedScope) throws IdentityOAuth2ScopeException
         return updatedScope;
     }
 
+    /**
+     * Get OAuth scope consent given for an application by the user.
+     *
+     * @param user      Authenticated user.
+     * @param appName   Application name.
+     * @param tenantId  Tenant Id.
+     * @return  {@link OAuth2ScopeConsentResponse}.
+     * @throws IdentityOAuth2ScopeConsentException
+     */
+    public OAuth2ScopeConsentResponse getUserConsentForApp(AuthenticatedUser user, String appName, int tenantId)
```

Comment:
Why OSGi api expects Authenticated user and resolves the userId from the osgin level? We can directly gets the userId in the osgilevel and resolves the userId from the authenticated in the upper level.

Reply:
The same applies for other apis as well

Reply:
Fixed